### PR TITLE
Implement defineFunctionEventProperties for event with multiple JSON schema definitions through the use of anyOf keywork

### PR DESCRIPTION
**Pull request by fredericbarthelet**
Closes: #8485

### CODE REVIEW

File: lib/classes/ConfigSchemaHandler/index.js [hunk 1]
```diff
@@ -232,7 +232,29 @@ class ConfigSchemaHandler {
       );
     }
 
-    addPropertiesToSchema(existingEventDefinition.properties[name], configSchema);
+    let definitionToUpdate;
+    if (
+      Object.prototype.hasOwnProperty.call(existingEventDefinition.properties[name], 'properties')
```
Comment (medikoo) [inline lib/classes/ConfigSchemaHandler/index.js:237]: Let's configure it as `if (existingEventDefinition.properties[name].type === 'object')`. I think it'll be more to the point

File: lib/classes/ConfigSchemaHandler/index.js [hunk 2]
```diff
@@ -232,7 +232,29 @@ class ConfigSchemaHandler {
...
+    ) {
+      // Event root definition is an object definition
+      definitionToUpdate = existingEventDefinition.properties[name];
+    } else if (
+      Object.prototype.hasOwnProperty.call(existingEventDefinition.properties[name], 'anyOf')
```
Comment (medikoo) [inline lib/classes/ConfigSchemaHandler/index.js:242]: Here let's do simple existence check `if (existingEventDefinition.properties[name].anyOf)`

File: lib/classes/ConfigSchemaHandler/index.js [hunk 3]
```diff
@@ -232,7 +232,29 @@ class ConfigSchemaHandler {
...
+    ) {
+      // Event root definition has multiple definitions. Finding the object definition
+      definitionToUpdate = existingEventDefinition.properties[name].anyOf.find(definition =>
+        Object.prototype.hasOwnProperty.call(definition, 'properties')
```
Comment (medikoo) [inline lib/classes/ConfigSchemaHandler/index.js:246]: Here let's also convert to `definition => definition.type === 'object'`

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @fredericbarthelet, it looks good! I've proposed some simplications though.

As you pointed I don't think we should consider `oneOf` variant, and for a time-being implementation as you proposed should cover all cases.

**Comment by fredericbarthelet** [comment]
@medikoo, I implemented your comments :) Thanks for your review

**Review by medikoo — APPROVED** [review]
Thank you @fredericbarthelet !

**Pull request merged**